### DISCUSSION TITLE
как правильно (более просто) собрать json массив для remote? как преобразовать данные в строку?

### DISCUSSION THREAD

**Discussion by nikidjan** [totumonline/totum-ru-issues-and-discussions#590] [Как это сделать?] [answered]
мне нужно сделать remote который отдает json:
{"name":["one","two","three"]}

my_field - поле тип "Селект" с множественным выбором в таб "klienty"

в самой таблице для теста делаю поле тип "Данные"
и в коде пишу

> =: selectRow(table: 'klienty';  field: 'my_field'; where: 'id' = #id; order: 'my_field' asc)

получаю в поле результат:

> {
>   "name": [
>     "one",
>     "two",
>     "three"
>   ]
> }

все супер! то что надо! но как бы собрать строку json в коде ремоута, и запихнуть в эту строку данные но как строку?
может как то можно преобразовать не сложным образом данные в строку?

**сейчас код ремоута:**

> =: strAdd (str: '{'; str: $my_str; str: '}')
> my_str: selectRow(table: 'klienty';  field: 'my_field'; where: 'id' = $comp_id ; order: 'my_field' asc)
> comp_id: select(table: 'klienty'; field: 'id'; where: 'get_param' = $#get[get_param]; where: 'status' = 1)

**результат get запроса:**

> {"error":"Параметр [[str]] должен быть типа строка."}

что логично, там же данные.

додумался только до того что бы использовать не seelctRow, а просто select, тогда получаю массив, и через listJoin самому уже собирать строку и т.д. но это такой длинный путь, подумал что может есть готовое решение что б сразу в "строку"?

**Comment by erpic-ru** [comment, marked as answer]
Да, в настройках ответа в списке Remotes выберите json и будет отдаваться строка
https://ru.docs.totum.online/api#type_modul

Ваш код, примерно, такой:

```
=: rowCreate( field: "name" = $selectlist)
 selectlist: ....
 
```

**Reply by nikidjan** [reply]
rowCreate - да, спасибо! именно это и нужно было!

**Reply by ttm-alexeypolunin** [reply]
О! я и забыл, что там есть автоматический превращатор ответа в строку json.

Если же в коде надо получить json как строку (что бы в str например использовать и есть еще варианты, когда например в post передать нужно строкой), то это делает функция jsonCreate. А обратное действие (из строки в многомерную структуру) jsonExtract.
